### ISSUE TITLE
when use big resolution with grin it is freezing

### ISSUE THREAD

**Issue by killroyka** [Orama-Interactive/Pixelorama#415]
**Pixelorama version:**

**OS/device including version:**

**Issue description:**

**Steps to reproduce:**

**Comment by OverloadedOrama** [comment]
Could you please provide more details? I am not sure what the problem is.

**Comment by killroyka** [comment]
when i use full HD resolution and grin my FPS  low(sorry for my English)

**Comment by Riteo** [comment]
What are you referring to when you say "grin"? Is it like a program, or something?

**Comment by killroyka** [comment]
Grid, sorry for my English again it is use with control G

**Comment by novhack** [comment]
What is your PC specification? I have no problem with 1x1 pixel grid enabled even on a huge sprite 1024x1024px.

**Comment by killroyka** [comment]
![image](https://user-images.githubusercontent.com/61899144/104379684-ded69100-553a-11eb-9426-8a92b9afc9af.png)
it is not poferfull but i think it is freezing too much when use full hd

**Comment by killroyka** [comment]
make a full hd surface and turn on grid, i think it is gonna freeze

**Comment by novhack** [comment]
FullHD is extremely huge for pixel art and for Pixelorama which is completely written in interpreted GDScript.
In such case I would suggest to rather use some other program than Pixelorama like Krita or Gimp.
Those are much more suitable for traditional drawing.

**Comment by Dancovich** [comment]
@novhack Although I completely agree with pixel art being the intent and so Pixelorama should be used for small resolution images and isn't intended to deal with full sized images, are there any plans to port some routines to either C# or GDNative to speed up the app?

If I create a 64x64 project, create a selection inside the canvas, use the pen, select the Filled circle brush, type 1000px as the size of the brush and drag the mouse to draw inside this selection a couple of times, the app hangs for several seconds or will even crash.

This isn't a realistic scenario and could even be fixed by not allowing you to type the size of the brush (the slider only goes up to 25px) but I don't believe it should be that slow to just paint a completely solid brush of 1000x1000.

This isn't related to the original post and I can open a new issue for that if needed, but I do believe there are areas where the app could be sped up a little to create a performance room for some edge cases.

Here are my specs for reference. It's not a beefy computer, but since the app is made in Godot I believe the canvas is already hardware accelerated, making my GPU more than enough for the task.

Intel Core i5 3570
8GB DDR3 Memory
NVidia Geforce GTX 1650
Windows 10 Pro 64bit

**Comment by zhengying** [comment]
Yes Pixelorama is very lag when canvas is big, even only 512 * 512 on my iMac, 
This core issue does not change, more features are also meaningless

**Comment by liyaowhen** [comment]
ik fr, pixelorama is laggy with high resolutions, like 1980 * 1080 for making pixel art background.

**Comment by jamius19** [comment]
I'm using Pixelorama 1.1.9 Stable on Windows 11.
What I experienced is that if I'm on RGBA mode with a custom brush, it works smoothly.
As soon as I switch to Indexed mode, the pencil starts to lag a lot.
